### PR TITLE
feat: [PROD-3870] Setup otelm and move tracer

### PR DESCRIPTION
**Pull request by Sansinj**
This PR implement a OTEL struct type which contains the tracer, it will allow in a next PR to add meter in the Otel struct.
- Move Tracer implementation in own folder + add test for the GRPC (only http was implemented which is not the one we use in atlas)
- Rename tracer specific config to allow implement of meter.

### CODE REVIEW

File: otelx/tracer.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	case f.AddCase("jaeger"):
+		tracer, prop, err := SetupJaegerTracer(c.Name, c)
+		if err != nil {
+			return err
+		}
+
+		t.tracer = tracer
+		t.propagator = prop
+		l.Infof("Jaeger tracer configured! Sending spans to %s", c.Providers.Jaeger.LocalAgentAddress)
+	case f.AddCase("otel"):
```
Comment (etiennecl) [inline otelx/tracer.go:33]: This should be kept as `otlp` OTEL is how we instrument the app although OTLP is the protocol used to export span

File: otelx/tracer.schema.json [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
-    "description": "Configure distributed tracing using OpenTelemetry",
+    "description": "Configure distributed tracer using OpenTelemetry",
     "properties": {
       "provider": {
         "type": "string",
-        "description": "Set this to the tracing backend you wish to use. Supports OTEL.",
-        "enum": ["otlp", "stdout", "jaeger"],
-        "examples": ["otlp"]
+        "description": "Set this to the tracer backend you wish to use. Supports OTEL.",
+        "enum": ["otel", "stdout", "jaeger"],
```
Comment (etiennecl) [inline otelx/tracer.schema.json:11]: Same

File: otelx/tracer.go (hunk 1 above)
Comment (Sansinj) [reply otelx/tracer.go:33]: I didn't change the name, it was already called `otel` in the previous `otel.go` but I agree `otlp` would be the right name

Comment (Sansinj) [reply otelx/tracer.go:33]: Update: Keep `otel` as reference to otel-collector

**Pull request merged**